### DISCUSSION TITLE
How to confirm the password?

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#5028] [Help]
This is the code, but the form again shows that the password and the confirmation password are different. What's the problem?

```php
Password::make('password')
    ->required()
    ->revealable()
    ->copyable()
    ->generatable()
    ->rules($this->passwordRules())
    ->dehydrateStateUsing(static fn (string $state): string => Hash::make($state))
    ->same('passwordConfirmation'),
Password::make('passwordConfirmation')
    ->required()
    ->revealable()
    ->dehydrated(false),
```

**Comment by altimusin** [comment]
TextInput::make('password')->password()->same('passwordConfirmation')

**Reply by ArtMin96** [reply]
> TextInput::make('password')->password()->same('passwordConfirmation')

The `->password()` is already specified in the Password field.

**Reply by danharrin** [reply]
Does it work with a text input though? The custom field may be causing the problem

**Comment by zepfietje** [comment]
The `password` field is getting dehydrated as a hashed value, while the confirmation contains the plain text. Might that be the cause?

**Reply by danharrin** [reply]
I don't think so
